src/quote/swap/adapters/zeroex_v2/decode-actions.ts

```diff
@@ -0,0 +1,109 @@
+
+import { EthAddress } from 'constants/addresses'
+import { SettlerActionsABI } from './abis/SettlerActions'
+import { convertFrom0xFeesToUniPool, getExchangeFrom0xSource } from './utils'
+
+import type { SwapDataV3 } from 'utils'
+import type { Hex } from 'viem'
+
+enum SettlerAction {
+  AerodromeV3 = 'AERODROMEV3',
+  UniswapV3 = 'UNISWAPV3',
+}
+
+// TBD if this is needed
+// function getMappedSource(source: string) {
+//   switch (source) {
+//     case 'Aerodrome_V3':
+//       return 'AERODROMEV3'
+//     case 'Uniswap_V3':
+//       return 'UNISWAPV3'
+//     case 'Uniswap_V2':
+//       return 'UNISWAPV2'
+//     case 'Sushiswap':
+//       return 'SUSHISWAP'
+//     case 'Balancer':
+//       return 'BALANCER'
+//     case 'Curve':
+//       return 'CURVE'
+//     case 'Kyber':
+//       return 'KYBER'
+//     case 'LiquidityProvider':
+//       return 'LIQUIDITYPROVIDER'
+//     case 'Mooniswap':
+//       return 'MOONISWAP'
+//     case 'MultiHop':
+//       return 'MULTIHOP'
+//     default:
+//       return source
+//   }
+// }
+
+function decodeUniPath(path: string): string[] {
+  const addressSize = 40
+  const tokenA = path.slice(0, addressSize + 2) // incl 0x
+  const tokenB = `0x${path.slice(path.length - addressSize, path.length)}`
+  return [tokenA, tokenB]
+}
+
+function isValidPath(
+  path: string[],
+  chainId: number,
+  inputToken: string,
+  outputToken: string,
+): boolean {
+  if (path.length < 2) return false
+  const weth = getTokenByChainAndSymbol(chainId, 'WETH')!.address
+  const inputTokenAddress = isAddressEqual(inputToken, EthAddress)
+    ? weth
+    : inputToken
+  if (!isAddressEqual(path[0], inputTokenAddress)) return false
+  const outputTokenAddress = isAddressEqual(outputToken, EthAddress)
+    ? weth
+    : outputToken
+  if (!isAddressEqual(path[path.length - 1], outputTokenAddress)) return false
+  return true
+}
+
+export function decodeActions(
+  actions: Hex[],
+  source: string,
+  chainId: number,
+  inputToken: string,
+  outputToken: string,
+): SwapDataV3 | null {
+  const actionsData = actions!.map((action: Hex) =>
+    decodeFunctionData({
+      abi: SettlerActionsABI,
+      data: action,
+    }),
+  )
+
+  const fees: number[] = []
+  let path: string[] = []
+  for (const action of actionsData) {
+    // Aerodome is returned as `UNISWAPV3` fork, so checking for the action would
+    // not work: `if (action.functionName === SettlerAction.AerodromeV3)`
+    if (action.functionName === SettlerAction.UniswapV3) {
+      // function UNISWAPV3(address recipient, uint256 bps, bytes memory path, uint256 amountOutMin) external
+      const [, bps, uniPath] = action.args
+      fees.push(convertFrom0xFeesToUniPool(bps))
+      path = decodeUniPath(uniPath)
+    }
+  }
+
+  if (!isValidPath(path, chainId, inputToken, outputToken)) {
+    console.error('Invalid path')
+    return null
+  }
+
+  return {
+    exchange: getExchangeFrom0xSource(source)!,
+    path,
+    fees,
+    pool: '',
```

Comment:
For the migration to all products we'll have to check if some still require a pool address or if it can be empty in general. Will do this on the cbBTC aerodrome PR.

Reply:
There is the additional property `tickSpacing` that will be added here - in the follow-up PR too.

Reply:
`pool` is only required for curve (which is only relevant for icETH)

(`path` and `fees`) are the only ones relevant for UniV3.